### PR TITLE
Fix Undefined Symbols while building llvm

### PR DESCRIPTION
**Pull request by anutosh491**
Not sure why the upload to Quetz failed on all 3 previous updates on llvm 

1) https://github.com/emscripten-forge/recipes/actions/runs/11376454739/job/31648960141
2) https://github.com/emscripten-forge/recipes/actions/runs/11209653508/job/31155192548
3) https://github.com/emscripten-forge/recipes/actions/runs/11052231541/job/30703847510

Need this for CppInterOp

### CODE REVIEW

**Comment by anutosh491** [comment]
Hmm, I thought the uploads failed but I see 19.1.2 here 
https://beta.mamba.pm/channels/emscripten-forge/packages/llvm#info

Am I missing something ?

**Comment by DerThorsten** [comment]
@anutosh491 the upload only fails client sided, but not on the server side!

**Comment by anutosh491** [comment]
Ahh cool. Needed llvm19 for CppInterOp which you already merged. Thanks for the clarification :)

**Comment by anutosh491** [comment]
Might just work out without the `-sERROR_ON_UNDEFINED_SYMBOLS=0` flag too. Giving it an attempt !

**Comment by anutosh491** [comment]
I realized the way i was passing the flags was hampering the symbols. The minor change in the last commit fixes most of it. It's only this one symbol that is undefined 
```
 │ │ wasm-ld: error: ../../../../lib/libclangInterpreter.a(Wasm.cpp.o): undefined symbol: lld::wasm::link(llvm::ArrayRef<char const*>, llvm::raw_ostream&, llvm:
 │ │ :raw_ostream&, bool, bool)
 ```
 Need to figure out how to fix this. If anyone has ideas, they are welcome !

**Comment by anutosh491** [comment]
Hey @DerThorsten I see you had added this through https://github.com/emscripten-forge/recipes/pull/595/commits/ec21b927427e4659b79bb3e9d0e8f86382a7811e
https://github.com/emscripten-forge/recipes/blob/a4c4a034d4f00b0aed0fcabbac477f3aa0e6558f/recipes/recipes_emscripten/llvm/build.sh#L7-L8

Maybe we won't like to have `WASM_BIGINT` but don't we need any others maybe like this
```
-s NO_INVOKE_RUN -s EXIT_RUNTIME -s INITIAL_MEMORY=64MB -s ALLOW_MEMORY_GROWTH -s EXPORTED_RUNTIME_METHODS=FS,callMain -s MODULARIZE -s EXPORT_ES6
```

Just curious if we need to have no `LDFLAGS` :\ ?

**Comment by anutosh491** [comment]
This is now ready. The above `link` function symbol is available in `lldWasm` and once the latest `Wasm.cpp` file was introduced in 19.0 (https://github.com/llvm/llvm-project/pull/86402) to help with running clang-repl and executing C++ code interactively in a wasm context ... we just need to link with lldWasm as the patch does.

File: recipes/recipes_emscripten/llvm/build.sh [hunk 1]
```diff
@@ -20,13 +20,12 @@ emcmake cmake ${CMAKE_ARGS} -S ../llvm -B .         \
...
     -DLLVM_ENABLE_LIBEDIT=OFF                       \
     -DLLVM_ENABLE_PROJECTS="clang;lld"              \
-    -DCMAKE_CXX_FLAGS="-Dwait4=__syscall_wait4"     \
     -DCMAKE_VERBOSE_MAKEFILE=ON                     \
     -DLLVM_ENABLE_THREADS=OFF                       \
-    -DCMAKE_CXX_FLAGS="-isystem $EMSCRIPTEN_FORGE_EMSDK_DIR/upstream/emscripten/cache/sysroot/include/c++/v1"
+    -DCMAKE_CXX_FLAGS="-Dwait4=__syscall_wait4 -isystem $EMSCRIPTEN_FORGE_EMSDK_DIR/upstream/emscripten/cache/sysroot/include/c++/v1"
 
 # Build step
-EMCC_CFLAGS='-sERROR_ON_UNDEFINED_SYMBOLS=0' emmake make -j1
```
Comment (IsabelParedes) [inline recipes/recipes_emscripten/llvm/build.sh:29]: :tada:

**Review by IsabelParedes — APPROVED** [review]

**Pull request merged**